crates/xmtp-networking/src/grpc_api_helper.rs

```diff
@@ -177,13 +178,16 @@ impl Client {
 pub struct Subscription {
     pending: Arc<Mutex<Vec<Envelope>>>,
     close_sender: Option<oneshot::Sender<()>>,
+    closed: Arc<AtomicBool>,
 }
 
 impl Subscription {
     pub async fn start(stream: Streaming<Envelope>) -> Self {
         let pending = Arc::new(Mutex::new(Vec::new()));
         let pending_clone = pending.clone();
         let (close_sender, close_receiver) = oneshot::channel::<()>();
+        let closed = Arc::new(AtomicBool::new(false));
```

Comment:
Took a bit of time to read up on AtomicBool. From what I can tell, Arc<AtomicBool> + clone is the right approach. You have my support :)